sdk/eventhub/Azure.Messaging.EventHubs/design/api-skeleton-january-2020.md

```diff
@@ -0,0 +1,535 @@
+    
+    public ProcessErrorEventArgs(
+        string partitionId,
+        string operation,
+        Exception exception,
+        CancellationToken cancellationToken = default);
+}
+
+public enum ProcessingStoppedReason 
+{
+    Shutdown,
+    OwnershipLost
+}
+```
+
+## Package: Azure.Messaging.EventHubs.Processor
+
+### `Azure.Messaging.EventHubs.Processor`
+```csharp
+public class EventProcessorClient
+{
+    public string FullyQualifiedNamespace { get; }
+    public string EventHubName { get; }
+    public string ConsumerGroup { get; }
+    public bool IsRunning { get; protected set; }
+    public string Identifier { get; }
+    
+    // Optional events
+    public event Func<PartitionInitializingEventArgs, Task> PartitionInitializingAsync;
+    public event Func<PartitionClosingEventArgs, Task> PartitionClosingAsync;
+
+    // Required events
+    public event Func<ProcessArgs, Task> ProcessEventAsync;
+    public event Func<ProcessErrorEventArgs, Task> ProcessErrorAsync;
+    
+    // Basic constructors
+    public EventProcessorClient(
+        BlobContainerClient checkpointStore,
+        string consumerGroup, 
+        string connectionString);
+        
+    public EventProcessorClient(
+        BlobContainerClient checkpointStore,
+        string consumerGroup, 
+        string connectionString, 
+        EventProcessorClientOptions clientOptions);
+        
+    public EventProcessorClient(
+        BlobContainerClient checkpointStore,
+        string consumerGroup, 
+        string connectionString, 
+        string eventHubName);
+        
+    public EventProcessorClient(
+        BlobContainerClient checkpointStore,
+        string consumerGroup, 
+        string connectionString, 
+        string eventHubName, 
+        EventProcessorClientOptions clientOptions);
+    
+    // Advanced constructors
+    public EventProcessorClient(
+        BlobContainerClient checkpointStore,
+        string consumerGroup,  
+        string fullyQualifiedNamespace, 
+        string eventHubName, 
+        TokenCredential credential, 
+        EventProcessorClientOptions clientOptions = null);
+    
+    // Mocking constructors
+    protected EventProcessorClient();
+    
+    // Behaviors
+    public virtual Task StartProcessingAsync(CancellationToken pcancellationToken = default);
+    public virtual void StartProcessing(CancellationToken cancellationToken = default);
+    public virtual Task StopProcessingAsync(CancellationToken cancellationToken = default);
+    public virtual void StopProcessing(CancellationToken cancellationToken = default);
+    
+    protected virtual Task OnPartitionInitializingAsync(PartitionInitializingEventArgs eventArgs);
+    protected virtual Task OnPartitionClosingAsync(PartitionClosingEventArgs eventArgs);
+    protected virtual Task OnProcessEventAsync(ProcessEventArgs eventArgs);
+    protected virtual Task OnProcessErrorAsync(ProcessErrorEventArgs eventArgs);
```

Comment:
I believe  `OnPartitionInitializingAsync()`, `OnPartitionClosingAsync()`, `OnProcessEventAsync()` and `OnProcessErrorAsync()`  methods are private.

Reply:
Good catch.   Those were indeed made `private` and should not appear.